File: src/main/frontend/src/app/models/dto/request/surveyRequest.ts
import {SpecialityResponse} from "../response/specialityResponse";
import {QuestionRequest} from "./questionRequest";
import {RecommendationRequest} from "./recommendationRequest";

export interface SurveyRequest {
  title: string,
  description: string,
  isRating: boolean,
  isPublic: boolean,
  speciality: SpecialityResponse | null,
  questions: QuestionRequest[],
  recommendations: RecommendationRequest[] | null,
  userId: number
}

File: src/main/backend/src/main/java/io/server/backend/dto/request/SurveyRequest.java
package io.server.backend.dto.request;

import io.server.backend.dto.response.SpecialityResponse;

import java.util.List;

public record SurveyRequest(
        String title,
        String description,
        Boolean isRating,
        Boolean isPublic,
        SpecialityResponse speciality,
        List<QuestionRequest> questions,
        List<RecommendationRequest> recommendations,

        Long userId
) { }
